### PR TITLE
Add missing proto definitions

### PR DESCRIPTION
**Pull request by matteojug**
## Description

Closes: https://github.com/stacks-network/sbtc/issues/846

## Changes

Add missing proto definitions and update some with the latest changes.

## Testing Information

Tested e2e on https://github.com/stacks-network/sbtc/pull/880, and via conversions in https://github.com/stacks-network/sbtc/pull/879.

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by djordon — COMMENTED** [review]
Largely looks good, had a few comments.

File: protobufs/stacks/signer/v1/requests.proto [hunk 1]
```diff
@@ -39,9 +39,30 @@ message StacksTransactionSignRequest {
...
+    // Ssmart contract deployment
+    SmartContract smart_contract = 10;
   }
 }
 
+enum SmartContract {
+  SC_UNSPECIFIED = 0;
+  // The sbtc-registry contract. This contract needs to be deployed
+  // before any other contract.
+  SBTC_REGISTRY = 1;
```
[the remark this answers is not in the document]
Comment (djordon) [reply protobufs/stacks/signer/v1/requests.proto:51]: Also, I thought that we checked for that convention as part of linting.

File: protobufs/crypto/common.proto [hunk 2]
```diff
@@ -43,12 +43,10 @@ message Scalar {
...
 // public key, like the crypto.PublicKey type, or the point at infinity.
+// This is represented as `Point` compressed bytes
 message Point {
-  // The x-coordinate of the point.
-  Uint256 x_coordinate = 1;
-  // Represents the parity bit of the point. True means the parity is
-  // odd, while false means the parity is even.
-  bool parity_is_odd = 2;
+  // The bytes of the point compressed representation
+  bytes data = 1;
```
[the remark this answers is not in the document]
Comment (matteojug) [reply protobufs/crypto/common.proto:49]: Given the (apparent?) generic nature of `Point` (vs `PublicKey`), I wasn't sure representing it as x + parity was enough. But no other reason that that, so if we can actually represent it like that happy to revert this.

[the remark this answers is not in the document]
Comment (djordon) [reply protobufs/crypto/common.proto:49]: I think the only difference between a `Point` and a `PublicKey` for the secp256k1 curve is that Zero is possible for a `Point` but not for a `PublicKey`.

[the remark this answers is not in the document]
Comment (matteojug) [reply protobufs/crypto/common.proto]: Reverted in [09113f2](https://github.com/stacks-network/sbtc/pull/878/commits/09113f2e10635e4547ecd9d064407b1c30e867fd)

File: protobufs/stacks/signer/v1/requests.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (matteojug) [reply protobufs/stacks/signer/v1/requests.proto]: Changed (and added the lint directive) in [09113f2](https://github.com/stacks-network/sbtc/pull/878/commits/09113f2e10635e4547ecd9d064407b1c30e867fd)

**Review by djordon — APPROVED** [review]
Looks good ✅ 🚢

**Pull request merged**